include/zephyr/drivers/gpio.h

```diff
@@ -1300,6 +1330,16 @@ static inline int z_impl_gpio_port_set_masked_raw(const struct device *port,
 	const struct gpio_driver_api *api =
 		(const struct gpio_driver_api *)port->api;
 
+	for (size_t i = 0; i < sizeof(mask) * 8; ++i) {
```

Comment:
probably could have declare `i` as `gpio_pin_t` to prevent the casting later

Reply:
Also maybe a good idea to wrap with `if (IS_ENABLED(CONFIG_TRACING_GPIO)) { ... }`